pkg/render/logstorage.go

```diff
@@ -586,6 +614,113 @@ func (es elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
 		volumes = es.dexCfg.RequiredVolumes()
 	}
 
+	var autoMountToken bool
+	if es.installation.CertificateManagement != nil {
+
+		// If certificate management is used, we need to override a mounting options for this init container.
+		initFSName := "elastic-internal-init-filesystem"
+		initFSContainer := corev1.Container{
+			Name:  initFSName,
+			Image: es.esImage,
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: ptr.BoolToPtr(true),
+				RunAsUser:  ptr.Int64ToPtr(0),
+			},
+			Args: []string{"bash", "-c", "/mnt/elastic-internal/scripts/prepare-fs.sh"},
```

Comment:
Does the configuration get merged with the default init container that's there, or is the init container overwritten completely by this? I ask because if it's overwritten completely you're missing the resource limits, but if it's merged then these arguments are the same as before so you probably don't need to use include them.

Reply:
I'm astounded by this insight, really nice. I've noticed that some things they merge, others they don't. I will verify this and deal with it if necessary before resolving this thread.

Reply:
Added the resources, thanks!